Import fails to set image when snapshot is used

Hi,

I'm logging this separately from #278 as I believe #278 is about using stock images that are no longer present.

This issue can be reproduced with version 1.24.1 of the provider like this:

* Create a snapshot image
* Have a Terraform config creating a server from the snapshot image
* Run the following commands:
```
terraform init
terraform plan -out plan
terraform apply plan
rm -rf .terraform
rm terraform.tfstate terraform.tfstate.backup .terraform.lock.hcl
terraform init
terraform import hcloud_server.<server-name> <server-id>
```

Now `terraform plan` will report the server as a resource that will be re-created on `apply`.

Setting `TF_LOG=INFO` for the `import` command yields the following error:

```
2021-02-10T02:10:23.246+0100 [INFO]  plugin.terraform-provider-hcloud_v1.24.1: 2021/02/10 02:10:23 [ERROR] setting state: image: '' expected type 'string', got unconvertible type 'int': timestamp=2021-02-10T02:10:23.246+0100
```

The following diff seems to fix this:

```diff
diff --git a/internal/server/resource.go b/internal/server/resource.go
index 40cda8a2..85527567 100644
--- a/internal/server/resource.go
+++ b/internal/server/resource.go
@@ -635,7 +635,7 @@ func setServerSchema(d *schema.ResourceData, s *hcloud.Server) {
                if s.Image.Name != "" {
                        d.Set("image", s.Image.Name)
                } else {
-                       d.Set("image", s.Image.ID)
+                       d.Set("image", fmt.Sprintf("%d", s.Image.ID))
                }
        }
 }
```

However, not having much experience with Terraform provider testing, I didn't feel confident enough to implement an acceptance test for this use case, so I am refraining from opening a PR.

Thanks!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## unreleased
 
+BUG FIXES:
+* `hcloud_server` resource: image had a wrong type (int instead of string) when a server was created from a snapshot
+
 ## 1.24.1 (February 04, 2021)
 
 BUG FIXES:

--- a/internal/server/resource.go
+++ b/internal/server/resource.go
@@ -635,7 +635,7 @@ func setServerSchema(d *schema.ResourceData, s *hcloud.Server) {
 		if s.Image.Name != "" {
 			d.Set("image", s.Image.Name)
 		} else {
-			d.Set("image", s.Image.ID)
+			d.Set("image", fmt.Sprintf("%d", s.Image.ID))
 		}
 	}
 }

